### PR TITLE
ovmfvartool: init at unstable-2021-06-16

### PR DESCRIPTION
**Pull request by baloo**
###### Description of changes

This brings ovmfvartool to manipulate ovmf_vars.fd files

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/ovmfvartool/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  postPatch = ''
+    # needed in build but /usr/bin/env is not available in sandbox
+    substituteInPlace ovmfvartool \
+      --replace "/usr/bin/env python3" "${pythonPkg.interpreter}"
+  '';
+
+  installPhase = ''
+    mkdir $out $out/bin
```
Comment (superherointj) [inline pkgs/applications/virtualization/ovmfvartool/default.nix:23]:
```suggestion
    mkdir -p $out/bin
```

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 189216` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ovmfvartool</li>
  </ul>
</details>

File: pkgs/applications/virtualization/ovmfvartool/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenvNoCC, fetchFromGitHub, python3 }:
+
+let
+  pythonPkg = python3.withPackages (p: with p; [ pyyaml ]);
+in stdenvNoCC.mkDerivation rec {
```
Comment (superherointj) [inline pkgs/applications/virtualization/ovmfvartool/default.nix:5]:
```suggestion
stdenvNoCC.mkDerivation rec {
```
Can you move `pythonPkg` to `postPatch` as it is only used there?

File: pkgs/applications/virtualization/ovmfvartool/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  installPhase = ''
+    mkdir $out $out/bin
+    install -m 755 ovmfvartool $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "Parse and generate OVMF_VARS.fd from Yaml";
+    homepage = "https://github.com/hlandau/ovmfvartool";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ baloo ];
```
Comment (RaitoBezarius) [inline pkgs/applications/virtualization/ovmfvartool/default.nix:31]: Can you add me as a maintainer please?

**Review by RaitoBezarius — APPROVED** [review]
Looks great to me, I am planning to do some QEMU UEFI revamps in NixOS tests infrastructure and this would help me a lot, I am willing to co-maintain it with you if you are okay with it. :)
(modulo other changes)

**Comment by superherointj** [comment]
Please, squash commits.

File: pkgs/applications/virtualization/ovmfvartool/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenvNoCC, fetchFromGitHub, python3 }:
+
+stdenvNoCC.mkDerivation rec {
+  name = "ovmfvartool";
+  version = "2021.06.16";
```
Comment (superherointj) [inline pkgs/applications/virtualization/ovmfvartool/default.nix:5]: Date separator should be "-" instead of ".".

**Review by superherointj — CHANGES_REQUESTED** [review]

File: pkgs/applications/virtualization/ovmfvartool/default.nix (hunk 4 above)
Comment (superherointj) [reply pkgs/applications/virtualization/ovmfvartool/default.nix:5]: Also should have "unstable-".

Comment (superherointj) [reply pkgs/applications/virtualization/ovmfvartool/default.nix]: Remember to update commit message with new version.

Comment (baloo) [reply pkgs/applications/virtualization/ovmfvartool/default.nix]: I did

Comment (superherointj) [reply pkgs/applications/virtualization/ovmfvartool/default.nix]: Looks good. Waiting CI.

File: pkgs/applications/virtualization/ovmfvartool/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenvNoCC, fetchFromGitHub, python3 }:
+
+stdenvNoCC.mkDerivation rec {
+  name = "ovmfvartool";
```
Comment (superherointj) [inline pkgs/applications/virtualization/ovmfvartool/default.nix:4]: I'm sorry. There was an error here. It should have been `pname` instead of `name`.

Comment (baloo) [reply pkgs/applications/virtualization/ovmfvartool/default.nix:4]: Thanks! and thanks @RaitoBezarius for the fix.

**Pull request merged**